### PR TITLE
Remove libgtest-dev from DependencyInstaller

### PR DESCRIPTION
**Pull request by vvbandeira**
This is installed from ./third-party/gtest/CMakeLists.txt

### CODE REVIEW

File: etc/DependencyInstaller.sh [hunk 1]
```diff
@@ -188,7 +188,6 @@ _installUbuntuPackages() {
         libpcre3-dev \
         python3-dev \
         libreadline-dev \
-        libgtest-dev \
```
Comment (oharboe) [inline etc/DependencyInstaller.sh:191]: Was there a library that should be uninstalled too?

Comment (vvbandeira) [reply etc/DependencyInstaller.sh:191]: Not sure, I looked at when the dependency was introduced and it was only this line. See 9964e664f on #3062.

Comment (oharboe) [reply etc/DependencyInstaller.sh:191]: See https://github.com/The-OpenROAD-Project/OpenROAD/discussions/3513#discussioncomment-6265721

Comment (vvbandeira) [reply etc/DependencyInstaller.sh:191]: This is the only reference for `gtest` that I found. @QuantamHD can you confirm that the `gtest` you suggested uninstalling was just `libgtest-dev`?

Comment (QuantamHD) [reply etc/DependencyInstaller.sh:191]: Yes. I think the other thing we might consider removing is the option of using the system Gtest in the cmake file. It seems like Gtest in general prefers to be included in your build rather that being a system dep.

Comment (vvbandeira) [reply etc/DependencyInstaller.sh:191]: Sounds good. I will remove and also test this in a fresh Ubuntu 20.04 VM to make sure this does not break Harsh's or Peter's systems.

File: third-party/gtest/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
-find_package(GTest)
 if(NOT GTest_FOUND)
```
Comment (gadfort) [inline third-party/gtest/CMakeLists.txt:1]: Why keep the if()? Seems like it should be removed if we are not checking for the GTest package anymore.

Comment (vvbandeira) [reply third-party/gtest/CMakeLists.txt]: Addressed on [9324a29](https://github.com/The-OpenROAD-Project/OpenROAD/pull/3514/commits/9324a299b638ca02fa8ff443663732fe2fa1f219).

**Review by maliberty — APPROVED** [review]

**Pull request merged**